Update ceph dashboard port (#833)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_migrating-mgr-from-controller-nodes.adoc

$ sudo systemctl restart iptables
----
+
[NOTE]

> Default dashboard port for ceph-mgr in greenfield deployment is 8443. With director-deployed Ceph, the default port is 8444 because the service ran on the controller node and it was necessary to use this port in order to avoid a conflict.For adoption (brownfield), dashboard port should be updated to 8443 in ceph-mgr config and firewall rules.

Reviewer: nit: `conflitct.For adoption` <- you missed a space

Author: done

> Default dashboard port for ceph-mgr in greenfield deployment is 8443. With director-deployed Ceph, the default port is 8444 because the service ran on the controller node and it was necessary to use this port in order to avoid a conflict.For adoption (brownfield), dashboard port should be updated to 8443 in ceph-mgr config and firewall rules.

Reviewer 2: ```suggestion
The default dashboard port for `ceph-mgr` in a greenfield deployment is 8443. With director-deployed {Ceph}, the default port is 8444 because the service ran on the Controller node, and it was necessary to use this port to avoid a conflict. For adoption, update the dashboard port to 8443 in the `ceph-mgr` configuration and firewall rules.
```

Author: done

> . Update dashboard port in ceph mgr config to 8443 using the following steps

Reviewer: Because this is a procedure, I think this section needs to be extended with a few more wording around the node where these commands are run (e.g. `Login to the first controller node (controller-0) and update the dashboard port ....`).
@klgill to advice more.

Reviewer: No is not important, technically cephadm works on the three controller nodes, but controller-0 (the first mon) is (historically) the bootstrap  node. I'm ok to simply ask to "Log in to controller-0".

Author: done

+
----

> $ sudo cephadm shell

Reviewer: should we split this into two steps?

1. Get a cephadm shell:
`command`
2. Run the following commands to update the ceph dashboard port and restart the ceph-mgr module:
`commands`

Reviewer 2: With the expanded wording above, I think it's okay to keep these commands as one step. However, I wonder what makes sense from a customer's perspective. Is it confusing for the these commands to remain in one step?

Reviewer: No, we can keep as a single step, it's fine from me. I was wondering if we should split but if it's ok for you to keep them as a single step no changes are required here.

and add the following content:
+
----
# 113 ceph_mgr {'dport': ['6800-7300', 8443]}

> add rule inet filter TRIPLEO_INPUT tcp dport { 6800-7300,8443 } ct state new counter accept comment "113 ceph_mgr"

Reviewer: -1: the `Ceph dashboard`, for a `Director` deployed `Ceph`, listens on `8444`. As per our conversation, and because 18 greenfield uses `8443`, we can add `8443` to make sure we're not going to have firewall issues, but if we remove `8444` here, then we need to `reconfigure` the Ceph dashboard to change port, and we shouldn't do that.

My proposal here is to have an additional `8443` to the port list, but not removing `8444`, which has reconfiguration implications.

Author: @fmount After we decided to go with option2 (switch from 8444 to 8443 during adoption),  i think this comment is no longer applicable

----
[NOTE]

> Default dashboard port for ceph-mgr in greenfield deployment is 8443, whereas the adoption from director deployed ceph needs the port to be changed from 8444 to 8443.

Reviewer: -1: we do not want to change port for an adopted environment. If that is the idea we should:

1. Document the steps required to reconfigure the dashboard on 8443 (remove the existing `ceph mgr config set ...`, disable and enable the ceph dashboard module) 
2. Update the automation (test suite) to test this scenario during the migration to validate this doc update.

The alternative here is to document that for pre-existing environments are keeping the same port, but 8443 is added because of greenfield.

Reviewer 3: I think we should change this:

> Default dashboard port for ceph-mgr in greenfield deployment is 8443, whereas the adoption from director deployed ceph needs the port to be changed from 8444 to 8443.

To this:

Default dashboard port for ceph-mgr in greenfield deployment is 8443. With director-deployed Ceph, the default port is 8444 because the service ran on the controller node and it was necessary to use this port in order to avoid a conflict.

We can make this change without deciding which option (1 or 2) to use as described above. My aim here was to capture this history in the doc since someone might ask and we can have the answer ready before they do.

Author: as we decided to go ahead with idea of changing the port  (8444 ->8443) in the adopted environment, 
- i have updated the doc with the steps to update the port in mgr dashboard config along with firewall rules
- i will work on the ceph_migration role and do the needful to match/validate the doc update.
